packages/build-tools/src/steps/functions/cache.ts

```diff
@@ -0,0 +1,104 @@
+import { BuildJob } from '@expo/eas-build-job';
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+} from '@expo/steps';
+
+import { CustomBuildContext } from '../../customBuildContext';
+
+export function createRestoreCacheBuildFunction(ctx: CustomBuildContext): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'restore-cache',
+    name: 'Restore Cache',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'key',
+        required: true,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'paths',
+        required: true,
+        defaultValue: [],
+        allowedValueTypeName: BuildStepInputValueTypeName.JSON,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'cache_key',
```

Comment:
nit: I think `key` should be sufficient, same for `paths`.

Or, we could not add it because however users are providing us `key` and `paths`, they should be able to use the same technique anywhere else? This is not a strong opinion.